### PR TITLE
Added option to create EFS or FSx Lustre file systems.

### PR DESCRIPTION
**Pull request by oleksii-mart**
Added new environment variable "deploy_filesystem_type". If it's set to "efs" or "fsx" then one of these filesystems will be created.

### CODE REVIEW

File: deploy/infra/aws/aws-native-infra/terraform/infra/input.tf [hunk 1]
```diff
@@ -75,6 +75,22 @@ variable "eks_additional_role_mapping" {
...
+  default     = 30
+  description = "Possible values are: 1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1096, 1827, 2192, 2557, 2922, 3288, 3653, and 0. If you select 0, the events in the log group are always retained and never expire."
+}
+
+###################################################################
+#                  Optional file systems
+###################################################################
+
+variable "deploy_filesystem_type" {
+  type        = string
```
Comment (SilinPavel) [inline deploy/infra/aws/aws-native-infra/terraform/infra/input.tf:94]: lets add validation here to check that only efs or fsx can be provided

File: deploy/infra/aws/aws-native-infra/terraform/infra/input.tf [hunk 2]
```diff
@@ -93,8 +109,38 @@ variable "efs_provisioned_throughput_in_mibps" {
   description = "EFS throughput, measured in MiB/s"
 }
 
-variable "eks_cloudwatch_logs_retention_in_days" {
+variable "fsx_s3_import_path" {
```
Comment (SilinPavel) [inline deploy/infra/aws/aws-native-infra/terraform/infra/input.tf]: we don't need this functionality

File: deploy/infra/aws/aws-native-infra/terraform/infra/input.tf [hunk 3]
```diff
@@ -93,8 +109,38 @@ variable "efs_provisioned_throughput_in_mibps" {
...
+  description = "The filesystem deployment type. One of: SCRATCH_1, SCRATCH_2, PERSISTENT_1, PERSISTENT_2"
+}
+
+variable "fsx_per_unit_storage_throughput" {
+  type        = number
+  default     = 200
+  description = "Describes the amount of read and write throughput for each 1 tebibyte of storage, in MB/s/TiB, required for the PERSISTENT_1 and PERSISTENT_2 deployment_type"
 }
+
+variable "fsx_subnet_id" {
```
Comment (SilinPavel) [inline deploy/infra/aws/aws-native-infra/terraform/infra/input.tf]: use the same subnets that where proided with `subnet_ids`

Comment (SilinPavel) [reply deploy/infra/aws/aws-native-infra/terraform/infra/input.tf]: if there is several subnets provided - use the first one

File: deploy/infra/aws/aws-native-infra/terraform/infra/input.tf [hunk 4]
```diff
@@ -93,8 +109,38 @@ variable "efs_provisioned_throughput_in_mibps" {
...
+  type        = list(string)
+  description = "A list of IDs for the subnets that the file system will be accessible from. File systems currently support only one subnet. The file server is also launched in that subnet's Availability Zone."
+}
+
+variable "aws_kms_key" {
```
Comment (SilinPavel) [inline deploy/infra/aws/aws-native-infra/terraform/infra/input.tf]: we already create several kms keys in this module we should use one of them

**Review by SilinPavel — CHANGES_REQUESTED** [review]

File: deploy/infra/aws/aws-native-infra/terraform/infra/input.tf [hunk 5]
```diff
@@ -10,12 +10,12 @@ variable "project_name" {
 
 variable "vpc_id" {
   type        = string
-  description = "Id of the VCP to be used for Compute Layer"
+  description = "Id of the VCP to be used for Cloud Pipeline"
 }
 
 variable "subnet_ids" {
   type        = list(string)
```
Comment (SilinPavel) [inline deploy/infra/aws/aws-native-infra/terraform/infra/input.tf:17]: add verification here that at least 1 subnet is provided

File: deploy/infra/aws/aws-native-infra/terraform/infra/input.tf [hunk 6]
```diff
@@ -93,8 +113,27 @@ variable "efs_provisioned_throughput_in_mibps" {
...
+  description = "The filesystem deployment type. One of: SCRATCH_1, SCRATCH_2, PERSISTENT_1, PERSISTENT_2"
+}
+
+variable "fsx_per_unit_storage_throughput" {
+  type        = number
+  default     = 200
+  description = "Describes the amount of read and write throughput for each 1 tebibyte of storage, in MB/s/TiB, required for the PERSISTENT_1 and PERSISTENT_2 deployment_type"
+}
+
+variable "fsx_subnet_id" {
```
Comment (SilinPavel) [inline deploy/infra/aws/aws-native-infra/terraform/infra/input.tf]: we can delete it now, right?

File: deploy/infra/aws/aws-native-infra/terraform/infra/input.tf [hunk 7]
```diff
@@ -75,6 +75,26 @@ variable "eks_additional_role_mapping" {
...
+#                  Optional file systems
+###################################################################
+
+variable "deploy_filesystem_type" {
+  type        = string
+  default     = "efs"
+  description = "Option to create EFS or FSx Lustre filesystem: must be set efs or fsx.If leave as is, neather will be created."
+  validation {
+    condition     = contains(["efs", "fsx"], var.deploy_filesystem_type)
+    error_message = "The value of the deploy_filesystem_type variable can be only efs or fsx. Please check if variable set correctly"
```
Comment (SilinPavel) [inline deploy/infra/aws/aws-native-infra/terraform/infra/input.tf]: check **that** variable **is** set correctly

Comment (oleksii-mart) [reply deploy/infra/aws/aws-native-infra/terraform/infra/input.tf]: Fixed

**Pull request merged**